M/mlpack/build_tarballs.jl

```diff
@@ -31,17 +32,15 @@ mkdir build && cd build
 # version.  So we'll just create a crappy little script, since Julia may not
 # be available in the build environment.
 echo "#!/bin/bash" > julia
-echo "echo \"Fake Julia version 1.3.0\"" >> julia
+echo "echo \"Fake Julia version 1.8.2\"" >> julia
 chmod +x julia
 
 FLAGS=(-DCMAKE_TOOLCHAIN_FILE=${CMAKE_TARGET_TOOLCHAIN}
+       -DCMAKE_CROSSCOMPILING=OFF
```

Comment:
Uhm, why this is needed?  We actually do cross-compilation (that's the entire magic of BinaryBuilder :sweat_smile:), so it's a bit weird that incorrectly setting `CMAKE_CROSSCOMPILING=OFF` makes it work (since this is false) :thinking:

Reply:
mlpack added some amount of support specific to cross-compiling that's built on top of CMake's cross-compiling support.  I didn't expect how that interacted with BinaryBuilder here (that gave [this error](https://github.com/JuliaPackaging/Yggdrasil/pull/5735#issuecomment-1289645105) that you pointed out), so I tried setting `CMAKE_CROSSCOMPILING` to `OFF` so that mlpack's cross-compiling CMake infrastructure didn't kick in.  It appears to work fine.

If you want to be a little more paranoid about it, I can instead use a patch on the `CMakeLists.txt` that forces the mlpack cross-compilation infrastructure to never be used, just let me know.

Reply:
That'd be nicer, but ok, let's go on with this one now since this is working.  Thanks!

Reply:
Awesome---thanks for the merge!  Now I'll update the mlpack.jl package (once `mlpack_jll` is pushed), and then hopefully (:crossed_fingers:) everything will work, otherwise I will be back with some other minor fix PR :smile:

Reply:
Thanks for keeping us updated @rcurtin! This is awesome! Will you ping me (here or elsewhere) when mlpack.jl is updated? We'd love to try things out at RAI with the updates. On a perhaps related note: a few people have upgraded to MacOS Ventura and are reporting issues with armadillo_jll not being able to load correctly. Not sure if the changes here will improve that, but I'm also not quite sure where to file an issue.

Reply:
@quinnj it is released now, so hopefully the update should work just fine.

I'm happy to take a quick look to try to help with any armadillo_jll issues.  However I agree, it's not clear precisely where to file an issue.  You could use https://github.com/mlpack/mlpack.jl, or, you could just send me an email at ryan@ratml.org.  The choice is yours. :)

Reply:
Great! I'll give it a whirl today.

For the armadillo_jll issue, I believe the BB wizards are discussing what is to be done in [this public Julia slack thread](https://julialang.slack.com/archives/C674ELDNX/p1667222083888139) if you're interested in following along; it sounds like Apple changed the linker behavior in Ventura.

Reply:
Thanks, following along there.  I'll jump into the conversation if I can add anything useful, but it looks like a lot of the hard digging has already been done.

Reply:
Don't hesitate to tag me if needed.

Reply:
😞 
```julia
  From worker 4:    signal (11): Segmentation fault
      From worker 4:    in expression starting at none:0
      From worker 4:    _ZN6mlpack13HoeffdingTreeINS_12GiniImpurityENS_24BinaryDoubleNumericSplitENS_25HoeffdingCategoricalSplitEE5TrainIN4arma11subview_colIdEEEEvRKT_m at /tmp/nix-build-raicode-tests.unit2.drv-0/rai-server-source/.julia/artifacts/5061b33ce5baf49f2784e1ab327aa4d27534afe0/lib/libmlpack_julia_hoeffding_tree.so (unknown line)
      From worker 4:    unknown function (ip: (nil))
      From worker 4:    Allocations: 4686818551 (Pool: 4684775049; Big: 2043502); GC: 294
Worker 4 terminated.
```

Reply:
I am unable to debug that without a way to reproduce.

Reply:
Yeah, I'll work on getting a smaller reproduction.